config.py.example

```diff
@@ -0,0 +1,122 @@
+#        #"uri" : "mongodb://user:passwd@localhost:27017", # mongodb std URI
+#        }
+#HUB_DB_BACKEND = {
+#        "module" : "biothings.utils.es",
+#        "host" : "localhost:9200",
+#        }
+
+# cache file format ("": ascii/text uncompressed, or "gz|zip|xz"
+CACHE_FORMAT = "xz"
+
+# Role, when master, hub will publish data (updates, snapshot, etc...) that
+# other instances can use (production, standalones)
+BIOTHINGS_ROLE = "slave"
+
+# Hub environment (like, prod, dev, ...)
+# Used to generate remote metadata file, like "latest.json", "versions.json"
+# If non-empty, this constant will be used to generate those url, as a prefix 
+# with "-" between. So, if "dev", we'll have "dev-latest.json", etc...
+# "" means production
+HUB_ENV = ""
+
+#* 2. Datasources *#
+ACTIVE_DATASOURCES = ConfigurationDefault(
+        default=[],
+        desc="List of package paths for active datasources")
+
+#* 3. Folders *# 
+# Path to a folder to store all downloaded files, logs, caches, etc...
+DATA_ARCHIVE_ROOT = ConfigurationError("Define path to folder which will contain all downloaded data, cache files, etc...")
+
+# where to store info about processes launched by the hub
+RUN_DIR = './run'
+
+# cached data (it None, caches won't be used at all)
+CACHE_FOLDER = None
+
+# Path to a folder to store all 3rd party parsers, dumpers, etc...
+DATA_PLUGIN_FOLDER = ConfigurationDefault(
+        default="./plugins",
+        desc="Define path to folder which will contain all 3rd party parsers, dumpers, etc...")
+
+# Path to folder containing diff files
+DIFF_PATH = ConfigurationDefault(
+        default=ConfigurationValue("""os.path.join(DATA_ARCHIVE_ROOT,"diff")"""),
+        desc="Define path to folder which will contain output files from diff")
+# Usually inside DATA_ARCHIVE_ROOT
+#DIFF_PATH = os.path.join(DATA_ARCHIVE_ROOT,"diff")
+
+# Path to folder containing release note files
+RELEASE_PATH = ConfigurationDefault(
+        default=ConfigurationValue("""os.path.join(DATA_ARCHIVE_ROOT,"release")"""),
+        desc="Define path to folder which will contain release files")
+
+# Usually inside DATA_ARCHIVE_ROOT
+#RELEASE_PATH = os.path.join(DATA_ARCHIVE_ROOT,"release")
+
+# this dir must be created manually
+LOG_FOLDER = ConfigurationDefault(
+        default=ConfigurationValue("""os.path.join(DATA_ARCHIVE_ROOT,"logs")"""),
+        desc="Define path to folder which will contain log files")
+# Usually inside DATA_ARCHIVE_ROOT
+#LOG_FOLDER = os.path.join(DATA_ARCHIVE_ROOT,'logs')
+                                                                                                                                                                                                    
+# default hub logger
+logger = ConfigurationDefault(
+        default=logging,
+        desc="Provide a default hub logger instance (use setup_default_log(name,log_folder)")
+# Usually use default setup
+#logger = setup_default_log("hub", LOG_FOLDER)
```

Comment:
Can we remove this block from `config.py`? We can move the logic to `ConfigurationWrapper`:

* if `LOG_FOLDER` is provided, but `logger` is not defined, we will create `logger` automatically.

Reply:
It's updated